src/H5T.c

```diff
@@ -3812,6 +3807,9 @@ H5T_decode(size_t buf_size, const unsigned char *buf)
     if (NULL == (f = H5F_fake_alloc((uint8_t)0)))
         HGOTO_ERROR(H5E_DATATYPE, H5E_CANTALLOC, NULL, "can't allocate fake file struct");
 
+    if (buf_size < 2)
```

Comment:
Not a big fan of an explicit comparison against the value 2 here. Using 2 occurrences of the `H5_IS_BUFFER_OVERFLOW` with a size of 1, for example, might be more verbose, but IMO is a lot clearer about what's going on.